src/main/java/Handler/PatchHandler.java

```diff
@@ -0,0 +1,57 @@
+import java.util.ArrayList;
+
+public class PatchHandler implements Handler {
+  private Directory directory;
+  private JsonPatchParser jsonPatchParser;
+  
+  public PatchHandler(Directory directory, JsonPatchParser jsonPatchParser) {
+    this.directory = directory;
+    this.jsonPatchParser = jsonPatchParser;
+  }  
+
+  public Response generateResponse(Request request) {
+    try {      
+      String requestContentType = request.getHeader(MessageHeader.CONTENT_TYPE);   
+      if (!this.directory.existsInStore(request.getURI())) {
+        return new Response.Builder(HttpStatusCode.NOT_FOUND)
+                           .build();
+      } else if (!requestContentType.equals(MimeType.JSON_PATCH)) {
+        return new Response.Builder(HttpStatusCode.UNSUPPORTED_MEDIA_TYPE)
+                           .setHeader(MessageHeader.ACCEPT_PATCH, MimeType.JSON_PATCH)
+                           .build();
+      }
+  
+      String updatedResourceContent = getUpdatedResourceContent(request);
+      this.directory.overwriteFileWithStringContent(request.getURI(), updatedResourceContent);
+      return buildOkResponse(request);
```

Comment:
There's a lot of checking/error handling happening in this method, and I think the OK response is sort of hard to find because of the conditional logic and because the response builder methods take up a lot of vertical space. I can't think of a good solution to this (you could extract 24 + 25 into a method called patch, but that is kind of silly, or you could extract some of the responses into methods but they are plenty readable as they are. On the other hand, you do use `buildOKResponse`). I assume this is something you noticed, but I think it's worth pointing out.